var/spack/repos/builtin/packages/cray-libsci/package.py

```diff
@@ -13,6 +13,8 @@ class CrayLibsci(Package):
     has_code = False    # Skip attempts to fetch source that is not available
 
     version("21.08.1.2")
+    version("20.10.1.2")
+    version("20.09.1.1")
```

Comment:
FWIW for external-only packages I wouldn't expect a need to register versions in `package.py` unless logic in the package file (or a dependent) was conditioned on those versions. Did you encounter an error somewhere without doing this?

Either way, adding versions here is fine, although can you move these to keep them in order?

Reply:
Thanks for that since I did wonder it becomes a bit of a nightmare of additions.  I will double check but I think the `modname` function causes some issues and its neater to have versions in Spack match the version of the code e.g. 

https://github.com/spack/spack/blob/986f927fffc8a5458c959d0fa38002ce493a5372/var/spack/repos/builtin/packages/cray-libsci/package.py#L46-L48

Maybe an as external package it will work but just will look odd where Spack says it has a version of the package which doesn't correspond to the version on the system.  Is there a version we can add such as `system` or `unknown` to make it clear it wouldn't match?

Reply:
> Maybe an as external package it will work but just will look odd where Spack says it has a version of the package which doesn't correspond to the version on the system.

It should be OK for an external config entry to specify a version that doesn't exist in the `package.py` file - or did that cause problems for you?